mantidproject/autoreduce
# 300 - Variable refactor

Fixes #122, #264, #299, #300

`InstrumentVariableUtils` has been cleaned up somewhat and many methods modified in how they work. The motivation was to hide all database-querying functions inside the utils, since view functions that did that work themselves just made a mess of things.

This made fixing the variable interfaces themselves relatively easy. #264 was a short fix; a good deal of the basic functionality was actually broken, and the whole interface should work as intended now.

To test:
* Go to an [instrument summary page](http://reduce.isis.cclrc.ac.uk/instrument/WISH/) on the production server
* Check that new variable sets can be created and saved as expected
* Check that existing variable sets can be edited as expected
* Check that if `tracks_script` is set, the variables can't be modified, and that they're kept up to date later on
* Check that deletion works as intended
* Check that text and list fields can be submitted empty
